frontend/src/__tests__/cypress/cypress/tests/e2e/pipelines/pipelines.cy.ts

```diff
@@ -0,0 +1,111 @@
+      AWS_S3_BUCKET: Buffer.from(AWS_BUCKETS.BUCKET_2.NAME).toString('base64'),
+      AWS_S3_ENDPOINT: Buffer.from(AWS_BUCKETS.BUCKET_2.ENDPOINT).toString('base64'),
+      AWS_SECRET_ACCESS_KEY: Buffer.from(AWS_BUCKETS.AWS_SECRET_ACCESS_KEY).toString('base64'),
+    };
+    createDataConnection(dataConnectionReplacements);
+
+    // Configure Pipeline server: Create DSPA Secret
+    const dspaSecretReplacements = {
+      DSPA_SECRET_NAME: dspaSecretName,
+      NAMESPACE: projectName,
+      AWS_ACCESS_KEY_ID: Buffer.from(AWS_BUCKETS.AWS_ACCESS_KEY_ID).toString('base64'),
+      AWS_SECRET_ACCESS_KEY: Buffer.from(AWS_BUCKETS.AWS_SECRET_ACCESS_KEY).toString('base64'),
+    };
+    createDSPASecret(dspaSecretReplacements);
+
+    // Configure Pipeline server: Create DSPA
+    const dspaReplacements = {
+      DSPA_SECRET_NAME: dspaSecretName,
+      NAMESPACE: projectName,
+      AWS_S3_BUCKET: AWS_BUCKETS.BUCKET_2.NAME,
+    };
+    createDSPA(dspaReplacements);
+  });
+
+  after(() => {
+    // Delete provisioned Project
+    deleteOpenShiftProject(projectName);
+  });
+
+  it('An admin User can Import and Run a Pipeline', () => {
+    // Login as an admin
+    cy.visitWithLogin('/', ADMIN_USER);
+
+    /**
+     * Import Pipeline by URL from Project Details view
+     */
+    projectListPage.navigate();
+    
+    // Open the project
+    filterProjectByName(projectName);
+    projectListPage.findProjectLink(projectName).click();
+
+    // Increasing the timeout to ~3mins so the DSPA can be loaded
+    projectDetails.findImportPipelineButton(180000).click();
+
+    // Fill tue Import Pipeline modal
+    pipelineImportModal.findPipelineNameInput().type(testPipelineName);
+    pipelineImportModal.findPipelineDescriptionInput().type('Pipeline Description');
+    pipelineImportModal.findImportPipelineRadio().click();
+    pipelineImportModal
+      .findPipelineUrlInput()
+      //TODO: modify this URL once the PR is merged
+      .type(
+        'https://raw.githubusercontent.com/opendatahub-io/odh-dashboard/caab82536b4dd5d39fb7a06a6c3248f10c183417/frontend/src/__tests__/resources/pipelines_samples/dummy_pipeline_compiled.yaml',
+      );
+    pipelineImportModal.submit();
+
+    //Verify that we are at the details page of the pipeline by checking the title
+    pipelineDetails.findPageTitle().should('have.text', testPipelineName);
+
+    /**
+     * Run the Pipeline using the Actions button in the pipeline detail view
+     */
+
+    pipelineDetails.selectActionDropdownItem('Create run');
+
+    //Fill the Create run fields
+    createRunPage.findExperimentSelect().click();
+    createRunPage.selectExperimentByName('Default');
+    createRunPage.fillName(testRunName);
+    createRunPage.fillDescription('Run Description');
+    createRunPage.findSubmitButton().click();
+
+    //Redirected to the Graph view of the created run
+    pipelineRunDetails.expectStatusLabelToBe('Succeeded');
```

Comment:
Default timeout is 10 seconds. Is that going to be sufficient while waiting for the run to complete?
Here's a scenario where it would be great to have some kind of conditional checks where if the status turns to error it fails.

I don't know how exactly to do this. But perhaps we note it for future improvements.

Reply:
It was waiting up to 3 minutes under the hood. But I have made it to be an argument here: [8c5da97](https://github.com/opendatahub-io/odh-dashboard/pull/3028/commits/8c5da97a11378ba311868612725265ba0bd9c2e1)